benchmarks/gluon_bench/reporter/reporter.go

```diff
@@ -0,0 +1,109 @@
+package reporter
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"time"
+
+	"github.com/bradenaw/juniper/xslices"
+
+	"github.com/ProtonMail/gluon/profiling"
+)
+
+type BenchmarkStatistics struct {
+	Total        time.Duration
+	Average      time.Duration
+	Fastest      time.Duration
+	Slowest      time.Duration
+	Median       time.Duration
+	Percentile90 time.Duration
+	Percentile10 time.Duration
+	RMS          time.Duration
+	SampleCount  int
+}
+
+func (b *BenchmarkStatistics) String() string {
+	return fmt.Sprintf("Total: %04dms SampleCount:%04d Fastest:%04dms Slowest:%04dms Average: %04dms Median: %04dms 90thPercentile: %04d ms 10thPercentile: %04dms RMS %04dms",
+		b.Total.Milliseconds(), b.SampleCount, b.Fastest.Milliseconds(), b.Slowest.Milliseconds(), b.Average.Milliseconds(),
+		b.Median.Milliseconds(), b.Percentile90.Milliseconds(), b.Percentile10.Milliseconds(), b.RMS.Milliseconds(),
+	)
+}
+
+func NewBenchmarkStatistics(durations ...time.Duration) BenchmarkStatistics {
+	sortedDurations := durations
+	sort.Slice(sortedDurations, func(i1, i2 int) bool {
+		return sortedDurations[i1] < sortedDurations[i2]
+	})
+
+	statistics := BenchmarkStatistics{}
+	statistics.SampleCount = len(sortedDurations)
+
+	if statistics.SampleCount == 1 {
+		statistics.Fastest = sortedDurations[0]
+		statistics.Slowest = sortedDurations[0]
+		statistics.Average = sortedDurations[0]
+		statistics.Total = sortedDurations[0]
+		statistics.Median = sortedDurations[0]
+		statistics.Percentile90 = sortedDurations[0]
+		statistics.Percentile10 = sortedDurations[0]
+		statistics.RMS = sortedDurations[0]
+	} else if statistics.SampleCount > 1 {
+		statistics.Fastest = sortedDurations[0]
+		statistics.Slowest = sortedDurations[statistics.SampleCount-1]
+		statistics.Total = xslices.Reduce(sortedDurations, 0, func(v1 time.Duration, v2 time.Duration) time.Duration {
+			return v1 + v2
+		})
+		statistics.Average = statistics.Total / time.Duration(statistics.SampleCount)
+		if statistics.Total%2 == 0 {
+			halfPoint := statistics.SampleCount / 2
+			statistics.Median = (sortedDurations[halfPoint] + sortedDurations[halfPoint+1]) / 2
+		} else {
+			statistics.Median = sortedDurations[((statistics.SampleCount + 1) / 2)]
+		}
+		statistics.Percentile90 = sortedDurations[int(math.Floor(float64(statistics.SampleCount)*(90.0/100.0)))]
+		statistics.Percentile10 = sortedDurations[int(math.Floor(float64(statistics.SampleCount)*(10.0/100.0)))]
+		var sumSquaredWithDiv float64
+		for i := 0; i < statistics.SampleCount; i++ {
+			// Dividing now rather than later or else we will trigger overflow.
+			f64Duration := float64(sortedDurations[i])
+			sumSquaredWithDiv += (f64Duration * f64Duration) / float64(statistics.SampleCount)
```

Comment:
sorry I described wrongly

```

sumsq := float64(0)
for  x := range durations {
  sumsq += x*x
}

rms = sqrt( sumsq / len(durations) )

```

Reply:
I did that initially, but the values get to high and it overflows the counter. We just do more divisions now, but the end result is the same.